hypershift-operator/controllers/nodepool/nodepool_controller_test.go

```diff
@@ -1324,6 +1378,275 @@ status: {}
+					},
+					Data: map[string]string{
+						tuningConfigKey: kubeletConfig1,
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bar",
+						Namespace: npNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: containerRuntimeConfig1,
+					},
+				},
+			},
+			existingConfigsInHcpNs: nil,
+			expectedMirroredConfigs: []corev1.ConfigMap{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      composeConfigName("foo", npName),
+						Namespace: hcpNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: kubeletConfig1,
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      composeConfigName("bar", npName),
+						Namespace: hcpNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: containerRuntimeConfig1,
+					},
+				},
+			},
+		},
+		{
+			name:                  "with configs that need to be deleted",
+			nodePool:              np,
+			controlPlaneNamespace: hcpNamespace,
+			configsToBeMirrored: []corev1.ConfigMap{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: npNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: kubeletConfig1,
+					},
+				},
+			},
+			existingConfigsInHcpNs: []client.Object{
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: npNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: kubeletConfig1,
+					},
+				},
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "bar",
+						Namespace: npNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: containerRuntimeConfig1,
+					},
+				},
+			},
+			expectedMirroredConfigs: []corev1.ConfigMap{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      composeConfigName("foo", npName),
+						Namespace: hcpNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: kubeletConfig1,
+					},
+				},
+			},
+			configsForDeletion: []corev1.ConfigMap{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      composeConfigName("bar", npName),
+						Namespace: hcpNamespace,
+					},
+					Data: map[string]string{
+						tuningConfigKey: containerRuntimeConfig1,
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			r := NodePoolReconciler{
+				Client:                 fake.NewClientBuilder().WithObjects(tc.existingConfigsInHcpNs...).Build(),
+				CreateOrUpdateProvider: upsert.New(true),
+			}
+			err := r.mirrorConfigs(context.Background(), logr.Discard(), tc.configsToBeMirrored, tc.controlPlaneNamespace, tc.nodePool)
+			g.Expect(err).ToNot(HaveOccurred())
+			for _, config := range tc.expectedMirroredConfigs {
+				cm := &corev1.ConfigMap{}
+				err := r.Get(context.Background(), client.ObjectKeyFromObject(&config), cm)
+				g.Expect(err).ToNot(HaveOccurred())
```

Comment:
Please use `cmp.Diff()` on the ConfigMap instead of these line-by-line assertions.

Reply:
Done.